### PR TITLE
Correctly set plot axes units to "code length" for datasets loaded with `unit_system="code"`

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

propose a fix for https://github.com/yt-project/yt/issues/2353
the fix is very simple but I'm not 100% confident it will not break anything else.

### CODE REVIEW

File: yt/visualization/plot_window.py [hunk 1]
```diff
@@ -864,8 +864,10 @@ def _setup_plots(self):
             axis_index = self.data_source.axis
 
             xc, yc = self._setup_origin()
-
-            if self._axes_unit_names is None:
+            if self.ds.unit_system.name.startswith("us"):
```
Comment (matthewturk) [inline yt/visualization/plot_window.py]: Why `"us"`?  Is that because it's short for "unit system"?

Comment (neutrinoceros) [reply yt/visualization/plot_window.py]: yes. That's the workaround I found. From the example I experimented with, `self.ds.unit_system.name` seems to start either with `"cgs_us"`, `"mks_us"` or just `"us"` , depending on which value is used to set the unit system at load time `yt.load(..., unit_system=<"cgs"|"mks"|"code">)`. If there are more possible cases I should try them out before we merge.

Comment (matthewturk) [reply yt/visualization/plot_window.py]: I don't think we need to do that!  I think this PR is probably good to go.

**Comment by neutrinoceros** [comment]
@yt-fido test this please.

**Pull request merged**